kratos/mpi/sources/mpi_data_communicator.cpp

```diff
@@ -474,6 +474,31 @@ bool MPIDataCommunicator::IsNullOnThisRank() const
     return mComm == MPI_COMM_NULL;
 }
 
+const DataCommunicator& MPIDataCommunicator::GetSubDataCommunicator(
+    const std::vector<int>& rRanks,
+    const std::string& rNewCommunicatorName
+    ) const
+{
+    if (ParallelEnvironment::HasDataCommunicator(rNewCommunicatorName)) {
+        const DataCommunicator& r_data_communicator = ParallelEnvironment::GetDataCommunicator(rNewCommunicatorName);
+        const int rank = Rank();
+        const auto it_find = std::find(rRanks.begin(), rRanks.end(), rank);
+        if (it_find != rRanks.end()) {
```

Comment:
this should be an error if it is not in the list

Reply:
The error is later if the data communicator is not null (it ay be null)